executor/executor.go

```diff
@@ -920,7 +920,7 @@ func (e *SelectionExec) Next(ctx context.Context, req *chunk.RecordBatch) error
 			if !e.selected[e.inputRow.Idx()] {
 				continue
 			}
-			if req.NumRows() >= req.Capacity() {
+			if req.IsFull() {
```

Comment:
at line 928, should we set the required rows based on the parent required rows?

Reply:
I think it's dangerous since the selectivity is unknown.
It increases times of calling child's `Next` to `RequiredRows / selectivity`, which may decrease the performance.